### PR TITLE
Update testing scripts

### PR DESCRIPTION
**Pull request by ndellingwood**
Load newer gcc with clang/9, clang/10 (needed for c++17 testing and newer)
Update slurm for voltrino
Add sogpu01

### CODE REVIEW

File: scripts/testing_scripts/test_all_sandia [hunk 1]
```diff
@@ -768,8 +804,8 @@ setup_env() {
   done
 
   if [ -e ${CM_ALL_SCRIPT_PATH}/update_lib.sh ]; then
-     echo "calling ${CM_ALL_SCRIPT_PATH}/update_lib.sh $MACHINE"
-     source ${CM_ALL_SCRIPT_PATH}/update_lib.sh $MACHINE
+     echo "calling ${CM_ALL_SCRIPT_PATH}/update_lib.sh $MACHINE $compiler"
+     source ${CM_ALL_SCRIPT_PATH}/update_lib.sh $MACHINE $compiler
```
Comment (masterleinad) [inline scripts/testing_scripts/test_all_sandia:808]: Where is the second argument used?

**Review by masterleinad — COMMENTED** [review]
Other than the one question this looks OK to me at first sight.

File: scripts/testing_scripts/update_lib.sh [hunk 2]
```diff
@@ -13,18 +13,36 @@ if [ "$1" = kokkos-dev ]; then
     module swap sems-gcc/4.9.3 sems-gcc/6.4.0
     module list
   fi
+  CLANGVER=$(clang --version | grep "clang version" | cut -d " " -f 3)
+  if [[ "${CLANGVER}" = 9.* ]] || [[ "${CLANGVER}" = 10.* ]]; then
+    # Newer gcc needed for c++ standard beyond c++14
+    module swap sems-gcc/5.3.0 sems-gcc/6.4.0
+    module list
+  fi
```
Comment (e10harvey) [inline scripts/testing_scripts/update_lib.sh:21]: Consider using a function.

File: scripts/testing_scripts/update_lib.sh [hunk 3]
```diff
@@ -13,18 +13,36 @@ if [ "$1" = kokkos-dev ]; then
...
   if [[ "${ICPCVER}" = 17.* ]]; then
     module swap sems-gcc/4.9.3 sems-gcc/6.4.0
     module list
   fi
+  CLANGVER=$(clang --version | grep "clang version" | cut -d " " -f 3)
+  if [[ "${CLANGVER}" = 9.* ]] || [[ "${CLANGVER}" = 10.* ]]; then
+    # Newer gcc needed for c++ standard beyond c++14
+    module swap sems-gcc/5.3.0 sems-gcc/6.4.0
+    module list
+  fi
```
Comment (e10harvey) [inline scripts/testing_scripts/update_lib.sh:34]: Same as above.

File: scripts/testing_scripts/update_lib.sh [hunk 4]
```diff
@@ -13,18 +13,36 @@ if [ "$1" = kokkos-dev ]; then
...
```
Comment (e10harvey) [inline scripts/testing_scripts/update_lib.sh:47]: Same as above.

**Review by e10harvey — APPROVED** [review]
Thanks, @ndellingwood ! I left a few comments below. If newer gcc versions are backwards compatible with c++14 or earlier and clang is loaded via a module, consider filing a ticket with the clang module maintainers to update the gcc version loaded by those clang modules.

File: scripts/testing_scripts/test_all_sandia (hunk 1 above)
Comment (ndellingwood) [reply scripts/testing_scripts/test_all_sandia:812]: I just pushed changes to the update_lib.sh using this argument, missed those changes in initial PR, thanks for the catch

File: scripts/testing_scripts/update_lib.sh (hunk 2 above)
Comment (ndellingwood) [reply scripts/testing_scripts/update_lib.sh]: Replaced by function in recent push, thanks for feedback

File: scripts/testing_scripts/update_lib.sh (hunk 3 above)
Comment (ndellingwood) [reply scripts/testing_scripts/update_lib.sh:25]: Replaced by function in recent push, thanks for feedback

File: scripts/testing_scripts/update_lib.sh (hunk 4 above)
Comment (ndellingwood) [reply scripts/testing_scripts/update_lib.sh]: Replaced by function in recent push, thanks for feedback

**Comment by ndellingwood** [comment]
@dalg24 @crtrott this should be ready for merge

File: scripts/testing_scripts/test_all_sandia [hunk 5]
```diff
@@ -112,6 +112,10 @@ if [[ "$HOSTNAME" == kokkos-dev\.sandia\.gov* ]]; then
   MACHINE=kokkos-dev
 fi
 
+if [[ "$HOSTNAME" == sogpu01.sandia.gov* ]]; then
```
Comment (dalg24) [inline scripts/testing_scripts/test_all_sandia:115]: I thought the point of the wildcard `*` that you don't need to spell out `sandia.gov`
Also you escaped the dot above for other machines.  Please confirm that you don't need to.

Comment (ndellingwood) [reply scripts/testing_scripts/test_all_sandia]: @dalg24 I updated the PR to remove .sandia.gov, thanks
cf060480bbc92f46900db2d563327fa3f7d94d9b

**Pull request merged**